walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,96 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.ListIterator;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    // Initiatize return data structure and set duration parameter
+    Collection<TimeRange> availTimes = new HashSet<>();
+    ArrayList<TimeRange> bookTimes = new ArrayList<TimeRange>();
+    final int length = (int) request.getDuration();
+    Collection<String>  requestAttendees = request.getAttendees();
+    
+    // Create a List of relevant events that will restrict available time ranges
+    bookTimes = findTimeConflicts(requestAttendees, events);
+
+    // Check that there are booked meetings before sorting it and running the available function
+    if (bookTimes.isEmpty()) {
+            availTimes.add(bookTimes.get(0).WHOLE_DAY);
+    }
+
+    else {
+        Collections.sort(bookTimes, bookTimes.get(0).ORDER_BY_START);
+        availTimes = availableTimes(bookTimes, length);
+    }
+
+    return availTimes;
+  }
+
+  public ArrayList<TimeRange> findTimeConflicts(Collection<String> requestAttendees, Collection<Event> events){
+    // Create a List of relevant events that will restrict available time ranges
+    ArrayList<TimeRange> bookTimes = new ArrayList<TimeRange>();
+
+    for (Iterator<Event> eventIterator = events.iterator(); eventIterator.hasNext();){
+        Event event = eventIterator.next();
+      
+        for (Iterator<String> attendIterator = requestAttendees.iterator(); attendIterator.hasNext();){
+            String attendee = attendIterator.next();
+           
+            if(event.getAttendees().contains(attendee) && !bookTimes.contains(event.getWhen())){
```

Comment:
The main reason I used an ArrayList was so I could sort it by time. But I initially made a HashSet to go through the for loop and then converted it to a list so it can be sorted.